### PR TITLE
clean up KTU and KeepConnections updates

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster I think this is what we want. thanks for the help.

looking at `KTUCommander.internKeepInUser`, I'm not sure we should be updated `addedAt` if a `ktu.state = active` exists. `addedAt` should reflect the first time the user was added, right?

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/AuthCommander.scala [hunk 1]
```diff
@@ -482,8 +482,9 @@ class AuthCommander @Inject() (
           db.readWrite { implicit s =>
             emailOpt.map(email => emailAddressCommander.saveAsVerified(UserEmailAddress.create(userId = userId, address = email)))
             addedByOpt.map { addedBy => // addedByOpt = None if no NonUserThread found for access token
-              log.info("[finishSignup] user has valid auth token, joining keep")
-              keepCommander.addUserToKeep(keepId, userId, Some(addedBy))
+              val keep = keepRepo.get(keepId)
+              ktuCommander.internKeepInUser(keep, userId, addedBy = addedBy)
+              keepRepo.save(keep.withConnections(keep.connections.plusUser(userId)))
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/AuthCommander.scala]: `keepCommander.persistKeep`

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/AuthCommander.scala [hunk 2]
```diff
@@ -466,13 +466,13 @@ class AuthCommander @Inject() (
...
-    log.info("[finishSignup] starting keep auto join")
     implicit val context = HeimdalContext.empty
     val hasPermission = db.readOnlyMaster(implicit s => permissionCommander.getKeepPermissions(keepId, Some(userId)).contains(KeepPermission.ADD_MESSAGE))
     if (hasPermission) {
-      log.info("[finishSignup] user has permission, joining keep")
       db.readWrite { implicit s =>
-        keepCommander.addUserToKeep(keepId, userId, addedBy = Some(userId))
+        val keep = keepRepo.get(keepId)
+        ktuCommander.internKeepInUser(keep, userId, addedBy = userId)
+        keepRepo.save(keep.withConnections(keep.connections.plusUser(userId)))
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/AuthCommander.scala]: `keepCommander.persistKeep`

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/AuthCommander.scala]: You can also check to see if `keep.connections.users.contains(userId)` first. Not sure how much that optimization is worth doing.

**Comment by ryanpbrewster** [comment]
Did you want to remove `keepCommander.addUserToKeep`?

**Comment by ryanpbrewster** [comment]
I don't disagree about updating `addedAt`. It seemed weird to me as well. Someone pointed out that if you, say, invite a user to participate in a discussion, you might expect to see them pop up to the top of the list even if they were already participating.

I do not personally find that use-case very compelling, but that's why we are updating `addedAt`.

**Comment by camhashemi** [comment]
if that's the only place where it's used, that's fine with me (where is addedBy used? should we be overwriting that?). I don't feel strongly, we'll probably revisit it as necessary.

looked at `addedBy`, looks like it's never being read. I'm guessing it's a remnant of what we had in eliza models.

**Pull request merged**